Guidelines.md

```diff
@@ -1998,12 +2007,100 @@ Content-Type: application/json
+Services SHOULD use verbose naming patterns and SHOULD NOT use abbreviations other than acronyms that are the dominant mode of expression in the domain being represented by the API, (e.g. Url).
+
+### 16.2 Casing
+- Acronyms SHOULD follow the casing conventions as though they were regular words (e.g. Url).
+- All identifiers including namespaces, entityTypes, entitySets, properties, actions, functions and enumeration values SHOULD use lowerCamelCase.
+- HTTP headers are the exception and SHOULD use standard HTTP convention of Capitalized-Hyphenated-Terms.
+
+### 16.3 Names to avoid
+Certain names are so overloaded in API domains that they lose all meaning or clash with other common usages in domains that cannot be avoided when using REST APIs, such as OAUTH.
+Services SHOULD NOT use the following names:
+- Context
+- Scope
+- Resource
+
+### 16.4 Forming compound names
+- Services SHOULD avoid using articles such as 'a', 'the', 'of' unless needed to convey meaning.
+	- e.g. names such as aUser, theAccount, countOfBooks SHOULD NOT be used, rather user, account, bookCount SHOULD be preferred.
+- Services SHOULD add a type to a property name when not doing so would cause ambiguity about how the data is represented or would cause the service not to use a common property name.
+- When adding a type to a property name, services MUST add the type at the end, e.g. createdDateTime.
+
+### 16.5 Identity properties
+- Services MUST use string types for identity properties.
+- For OData services, the service MUST use the OData @id property to represent the canonical identifier of the resource.
+- Services MAY use the simple 'id' property to represent a local or legacy primary key value for a resource.
+- Services SHOULD use the name of the relationship postfixed with 'Id' to represent a foreign key to another resource, e.g. subscriptionId.
+	- The content of this property SHOULD be the canonical ID of the referenced resource.
+
+### 16.6 Date and time properties
+
+- For properties requiring both date and time, services MUST use the suffix 'DateTime'.
+- For properties requiring only date information without specifying time, services MUST use the suffix 'Date', e.g. birthDate.
+- For properties requiring only time information without specifying date, services MUST use the suffix 'Time', e.g. appointmentStartTime.
+
+### 16.7 Name properties
+- For the overall name of a resource typically shown to users, services MUST use the property name 'displayName'.
+- Services MAY use other common naming properties, e.g. givenName, surname, signInName.
+
+### 16.8 Collections and counts
+- Services MUST name collections as plural nouns or plural noun phrases using correct English.
+- Services MAY use simplified English for nouns that have plurals not in common verbal usage.
+	- e.g. schemas MAY be used instead of schemata.
+- Services MUST name counts of resources with a noun or noun phrase suffixed with 'Count'.
+
+### 16.9 Common property names
```

Comment:
Sure, I see where you are coming from. However, I think clients not inspecting the contract is what make So many systems brittle and tightly coupled. The idea of *Out of Band Communication* about the api creates individual knowledge silos about how particular parts of the system work / have to work. The entire ecosystem around Swagger/OpenAPI and OData is about consuming & trusting the schema contract. I agree with you, many clients do not, but many clients **should**.

I think what really prompted me to insert myself into the conversation more than anything else was the type inference on a field name, but only on the `date` & `datetime` fields. It is very inconsistent within the other naming conventions - nothing else has a type. At the very least, maintain the convention within it self. All fields should include the type, or they should not include a type. I'm in favor of the former.